### ISSUE TITLE
Matrix multiplication order

### ISSUE THREAD

**Issue by michael** [STRd6/matrix.js#1]
Absolutely not sure if this is wrong but I'm a bit confused about the matrix multiplication order in Matrix#scale, Matrix#translate, Matrix#rotate

```
scale: function(sx, sy) {
    return Matrix.scale(sx, sy).concat(this);
},
```

If we take the scale for example, why is the newly generated scale matrix multiplied with the current one and not the other way round? 

```
scale: function(sx, sy) {
    return this.concat(Matrix.scale(sx, sy));
},
```

If I apply a series of transformations to an identity matrix I'd tend to read it from left to right.

```
var mat = new Matrix();
// Scaling around a reference point
mat.translate(5,5).scale(2,2).translate(-5,-5)
```

I'd suppose the matrix to be translated about 5,5 then scaled with 2,2 and finally translated back about -5,-5. But this results in an execution the other way round. To get the desired effect I'd have to flip the calling order. 

```
mat.translate(-5,-5).scale(2,2).translate(5,5)
```

If you look at the Matrix implementation of Processing.js, you'll see that they are doing it differently.  http://github.com/michael/unveil/blob/master/src/scene/matrix2d.js#L60 Not sure what's the right way. :)

Btw,maybe you should consider a aboutPoint parameter for scale() as well (as you do with rotate() already)

**Comment by STRd6** [comment]
Hey, thanks for the feedbak! I think you're right about the ordering of the concatenation. Also, I think the comments claimed "this matrix multiplied by a ..." so in either case at least something would need changing. The transformation methods should read the same as if you were to manually concatenate the matrices. 

Here's the test case I constructed to illustrate this:

<pre>
var m1 = Matrix().translate(tx, ty).scale(s).rotate(theta);
var m2 = Matrix().concat(Matrix.translation(tx, ty)).concat(Matrix.scale(s)).concat(Matrix.rotation(theta));

matrixEqual(m1, m2);
</pre>


I also added an aboutPoint parameter for scaling.
